src/main/java/com/deavensoft/paketomat/dispatcher/DispatcherServiceImpl.java

```diff
@@ -1,27 +1,142 @@
 package com.deavensoft.paketomat.dispatcher;
+
+import com.deavensoft.paketomat.center.model.*;
+import com.deavensoft.paketomat.center.model.Package;
+import com.deavensoft.paketomat.email.EmailDetails;
+import com.deavensoft.paketomat.email.EmailService;
+import com.deavensoft.paketomat.exceptions.NoSuchUserException;
+import com.deavensoft.paketomat.user.UserService;
+import lombok.AllArgsConstructor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
-import java.util.List;
-import java.util.Optional;
+import java.io.IOException;
+import java.util.*;
 
+@AllArgsConstructor
 @Service
 public class DispatcherServiceImpl implements DispatcherService {
-    private final DispatcherRepository dispatcherRepository;
+    private DispatcherRepository dispatcherRepository;
+    @Value("${spring.external.api.distance.url}")
+    private String url;
+    @Value("${spring.external.api.distance.header.host.name}")
+    private String hostName;
+    @Value("${spring.external.api.distance.header.host.value}")
+    private String hostValue;
+    @Value("${spring.external.api.distance.header.key.name}")
+    private String keyName;
+    @Value("${spring.external.api.distance.header.key.value}")
+    private String keyValue;
+    private UserService userService;
+    private EmailService emailService;
+
+    private  static final int MIN_POPULATION = 10000;
+    private  static final int SIZE_OF_PAKETOMAT = 5;
```

Comment:
Can this be in the configuration file too?

Reply:
We added it